src/app/public/modules/checkbox/checkbox-required-validator.directive.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  Directive,
+  forwardRef,
+  Provider
+} from '@angular/core';
+
+import {
+  CheckboxRequiredValidator,
+  NG_VALIDATORS
+} from '@angular/forms';
+
+// tslint:disable:no-forward-ref
+export const SKY_CHECKBOX_REQUIRED_VALIDATOR: Provider = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => SkyCheckboxRequiredValidatorDirective),
+  multi: true
+};
+// tslint:enable
+
+/**
+ * Validator support for the `required` input on `sky-checkbox`.
+ * Angular's `CheckboxRequiredValidator` only works with `input type=checkbox`. This directive
+ * extends the angular validator to work with `sky-checkbox` tags.
+ */
+@Directive({
+  selector: `sky-checkbox[required][formControlName],
+             sky-checkbox[required][formControl], sky-checkbox[required][ngModel]`,
+  providers: [SKY_CHECKBOX_REQUIRED_VALIDATOR]
+})
+export class SkyCheckboxRequiredValidatorDirective extends CheckboxRequiredValidator {}
```

Comment:
Some background -

The community [filed an issue](https://github.com/angular/angular/issues/11459) that checkboxes aren't `invalid` if required but unchecked. Angular's answer to this was to create a [separate validator just for checkboxes](https://angular.io/api/forms/CheckboxRequiredValidator).

However, since we are implementing our own custom checkbox control we have to add more selectors to the directive in order for it to work. See [Angular Material's implementation].(https://github.com/angular/components/blob/a415b52ead2ec202ba37edb7d7866b29dd790394/src/material/checkbox/checkbox-required-validator.ts)

Reply:
Thanks for the links! Do you mind putting one of them in the comment above the `Directive` decorator for future reference?

Reply:
Done.